src/tablet/tablet_impl.cc

```diff
@@ -2250,10 +2297,12 @@ void TabletImpl::SetExpire(RpcController* controller, const ::openmldb::api::Set
         response->set_msg("table is not exist");
         return;
     }
-    MemTable* mem_table = dynamic_cast<MemTable*>(table.get());
-    if (mem_table != NULL) {
-        mem_table->SetExpire(request->is_expire());
-        PDLOG(INFO, "set table expire[%d]. tid[%u] pid[%u]", request->is_expire(), request->tid(), request->pid());
+    if (table->GetStorageMode() == ::openmldb::common::StorageMode::kMemory) {
```

Comment:
logging error message and send back error code.

Add back this capability when doing disk table `gc`